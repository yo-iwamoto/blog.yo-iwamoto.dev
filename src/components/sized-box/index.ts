export { SizedBox } from "./sized-box";
