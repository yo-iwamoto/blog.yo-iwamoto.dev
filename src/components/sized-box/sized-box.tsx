import { cn } from "#src/lib/utils";
import type { ElementType, PropsWithChildren } from "react";

type Props = PropsWithChildren & {
  as?: ElementType;
  className?: string;
};

export function SizedBox({ children, as, className }: Props) {
  const Tag = as ?? "div";
  return (
    <Tag className={cn("max-w-4xl mx-auto px-4 lg:px-0", className)}>
      {children}
    </Tag>
  );
}
