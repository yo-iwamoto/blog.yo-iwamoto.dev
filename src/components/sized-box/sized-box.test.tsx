import { render, screen } from "@testing-library/react";
import { SizedBox } from "./sized-box";

describe("SizedBox", () => {
  it("children が描画されること", () => {
    const { getByText } = render(<SizedBox>children</SizedBox>);

    expect(getByText("children")).toBeInTheDocument();
  });

  it("as で指定したタグで描画されること", () => {
    render(<SizedBox as="h1">box</SizedBox>);

    expect(
      screen.getByRole("heading", { level: 1, name: "box" }),
    ).toBeDefined();
  });
});
