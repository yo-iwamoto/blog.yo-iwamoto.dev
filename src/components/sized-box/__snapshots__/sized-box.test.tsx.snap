// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SizedBox > snapshot unchanged 1`] = `
<DocumentFragment>
  <div
    class="max-w-4xl mx-auto px-4 lg:px-0"
  />
</DocumentFragment>
`;
